src/librustdoc/html/markdown.rs

```diff
@@ -631,7 +640,11 @@ impl LangString {
                     seen_rust_tags = seen_other_tags == false;
                 }
                 "no_run" => { data.no_run = true; seen_rust_tags = !seen_other_tags; }
-                "ignore" => { data.ignore = true; seen_rust_tags = !seen_other_tags; }
+                "ignore" => { data.ignore = Ignore::All; seen_rust_tags = !seen_other_tags; }
+                x if enable_per_target_ignores && x.starts_with("ignore-") => {
```

Comment:
I'm concerned that this forces people to send `--enable-per-target-ignores` even when building docs regularly. Rustdoc uses the `LangString` struct even during docs generation to see whether it should syntax-highlight a doctest, and also to display the little banners that show on `no_run` and `ignore` doctests. This means that the way this is written right now, adding `ignore-something` to a doctest and *not* sending `--enable-per-target-ignores` will set `seen_other_tags` below, and cause a doctest to show up as plain-text. One way to fix this would be to always handle `ignore-something` lang strings, and ignore them if `--enable-per-test-ignores` isn't set:

```suggestion
                x if x.starts_with("ignore-") => if enable_per_target_ignores {
```

Reply:
This is a fair point, and your change would prevent that problem. To also bring in something you mentioned in your main comment, right now rustdoc does not test anything with an invalid filter. So running `rustdoc --test` with an older version of rustdoc has the effect of silently ignoring any doctests with unknown (ie newer) filters.